hooks/pre-commit

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+
+STAGED_PY_FILES=$(git diff --cached --name-only --diff-filter=ACM | grep ".py$")
+STAGED_RS_FILES=$(git diff --cached --name-only --diff-filter=ACM | grep ".rs$")
+
+
+if [[ ("$STAGED_PY_FILES" = "") && ("$STAGED_RS_FILES" = "") ]]; then
```

Comment:
Can we split this into only running the rust linter only when rust files were touched and pylint only when python was touched.

Reply:
The linter already only runs on python files that are were changed as part of the commit. Unfortunately, I have not been able to run clippy on individual files (so certainly not only staged files). Best I can do so far is to avoid running `clippy` if no rust files were touched. HOw does that sound?

Reply:
Check out the new and improved pre-push!

```
➜  dex-services git:(hooks) git push
Skipping cargo test; Branch does not differ from master in a rusty way!
Skipping pytest; Branch does not differ from master in a pythonic way!
PUSH SUCCEEDED
```

Reply:
Yes, I think this is okay. Especially, since clippy seems to have its own cache and is not running on all files.

Reply:
nit: I’d suggest to remove this if/else branch altogether now since it feels unnecessary (covered by if and loop below)

Reply:
Yes, I could do that I just thought it would break earlier if we left it here... I have no personal preference.